main/lsp/LSPIndexer.cc

```diff
@@ -198,14 +200,7 @@ LSPIndexer::getTypecheckingPathInternal(const vector<shared_ptr<core::File>> &ch
     // foreground..." operation, we also compute how many downstream files (outside of the changed
     // files) would need to be typechecked on the fast path so we can compare that number against
     // `lspMaxFilesOnFastPath` as well.
-
-    // TODO(jez) Currently we compute the full set of information that we would need for the sake of
-    // whether to take the fast path twice--once here and once again in runFastPath on the
-    // typechecking thread.
-    //
-    // As an optimization, we might want to try to store that information on the update itself, so
-    // that the typechecking thread can simply read it instead of having to compute it.
-    auto result = LSPFileUpdates::fastPathFilesToTypecheck(*initialGS, *config, changedFiles, evictedFiles);
+    result = LSPFileUpdates::fastPathFilesToTypecheck(*initialGS, *config, changedFiles, evictedFiles);
```

Comment:
This is a bit flipped from how I would have structured it, I generally don't love out parameters (in/out parameters I find a little better).

Another way to do this would be to thread the `LSPFileUpdates` in to `getTypeCheckingPathInternal`. It would read `edit.updatedFiles`, populate `edit.fastPathExtraFiles`, populate `edit.typecheckingPath`, and simplify some call sites where we're currently passing in both `edit.updatedFiles` and also `FastPathFilesToTypecheckResult`.

It's probably fine as written, it just showed up kind of backwards from how it was going to work in my head.

Reply:
I started down that path initially, but there is a call site of `getTypecheckingPathInternal` that doesn't have a `LSPFileUpdates` value available to thread through: the version that only accepts a list of changed files, and is called from `SorbetWorkspaceEditTask`. That's called from a context where the `LSPFileUpdates` value cached on the workspace edit is explicitly known to be `nullptr`, but it might make sense to populate it in that case so that we always have one available?

Reply:
I reworked this so that we can return the file info along with the path, instead of populating an outparam.